pandas/core/dtypes/cast.py

```diff
@@ -1018,6 +1020,80 @@ def soft_convert_objects(
     return values
 
 
+def convert_dtypes(
+    input_array,
+    convert_string: bool = True,
+    convert_integer: bool = True,
+    convert_boolean: bool = True,
+) -> Dtype:
```

Comment:
we really need to get a DtypeObject in pandas._typing that excludes strings

Reply:
PR welcome!  (heh, heh)